sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/bitmapExpressions.scala

```diff
@@ -0,0 +1,318 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst.expressions
+
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.catalyst.analysis.TypeCheckResult
+import org.apache.spark.sql.catalyst.expressions.aggregate.ImperativeAggregate
+import org.apache.spark.sql.catalyst.expressions.codegen.CodegenFallback
+import org.apache.spark.sql.catalyst.trees.UnaryLike
+import org.apache.spark.sql.types.{AbstractDataType, BinaryType, DataType, LongType, StructType}
+
+/** Shared constants and methods for bitmap functions. */
+object BitmapFunctions {
+  /** Number of bytes in a bitmap. */
+  final val NUM_BYTES = 4 * 1024
+
+  /** Number of bits in a bitmap. */
+  final val NUM_BITS = 8 * NUM_BYTES
+
+  /** Merges both bitmaps and writes into bitmap1. */
+  def merge(bitmap1: Array[Byte], bitmap2: Array[Byte]): Unit = {
+    for (i <- 0 until Math.min(bitmap1.length, bitmap2.length)) {
+      bitmap1.update(i, ((bitmap1(i) & 0x0FF) | (bitmap2(i) & 0x0FF)).toByte)
+    }
+  }
+}
+
+@ExpressionDescription(
+  usage = "_FUNC_(child) - Returns the bucket number for the given input child expression.",
+  examples = """
+    Examples:
+      > SELECT _FUNC_(123);
+       1
+      > SELECT _FUNC_(0);
+       0
+  """,
+  since = "3.5.0",
+  group = "misc_funcs"
+)
+case class BitmapBucketNumber(child: Expression)
+  extends UnaryExpression with ImplicitCastInputTypes with NullIntolerant with CodegenFallback {
```

Comment:
It's more recommended to implement an expression with a static java method. See `AesEncrypt` as an example.

Reply:
I updated these expressions to use `StaticInvoke()`.